extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/binder/vertx/VertxHttpServerMetrics.java

```diff
@@ -148,12 +178,18 @@ public void responseEnd(HttpRequestMetric requestMetric, HttpResponse response,
                 config.getServerIgnorePatterns());
         if (path != null) {
             Timer.Sample sample = requestMetric.getSample();
-            Timer.Builder builder = Timer.builder(nameHttpServerRequests)
-                    .tags(Tags.of(
-                            VertxMetricsTags.method(requestMetric.request().method()),
-                            HttpCommonTags.uri(path, response.statusCode()),
-                            VertxMetricsTags.outcome(response),
-                            HttpCommonTags.status(response.statusCode())));
+            Tags allTags = Tags.of(
+                    VertxMetricsTags.method(requestMetric.request().method()),
+                    HttpCommonTags.uri(path, response.statusCode()),
+                    VertxMetricsTags.outcome(response),
+                    HttpCommonTags.status(response.statusCode()));
+            if (!additionalHttpServerMetricsTagsProviders.isEmpty()) {
+                AdditionalHttpServerMetricsTagsProvider.Context context = new DefaultContext(requestMetric.request());
+                for (int i = 0; i < additionalHttpServerMetricsTagsProviders.size(); i++) {
```

Comment:
The reason for using foreach here is that I want to avoid creating garbage on this path